File: jinja-mako-benchmark/hello.py
import tornado.httpserver
import tornado.web
import tornado.ioloop
import asyncmongo
import render

db = asyncmongo.Client(pool_id='x', dbname='test', host='localhost', port=27017)

class Index(tornado.web.RequestHandler):
    @tornado.web.asynchronous
    def get(self):
        db.posts.find(dict(), callback=self.on_result)

    def on_result(self, result, error):
        if error:
            raise tornado.web.HTTPError(500)
        self.write(render.render('index.html', posts=result))
        self.finish()

def main():
    application = tornado.web.Application([
            (r'/', Index),
        ],
    )
    tornado.httpserver.HTTPServer(application).listen(8888)
    tornado.ioloop.IOLoop.instance().start()

if __name__ == '__main__': main()

File: jinja-mako-benchmark/hello.js
var jinjs = require('jinjs');
jinjs.registerExtension('.html');
var index = require('./jinja/index');
var db = require('mongojs')('test');

require("http").createServer(function(request, response) {
    db.collection('posts').find(function(err, posts) {
        if (err) {
            response.writeHead(500);
        } else {
            response.writeHead(200, {'Content-Type': 'text/html'});
            response.write(index.render({ posts: posts }));
        }
        response.end();
    });
}).listen(8888);
